File: app-java/src/test/java/denshchikov/dmitry/taskcreator/service/TaskServiceTest.java
package denshchikov.dmitry.taskcreator.service;

import denshchikov.dmitry.taskcreator.config.IntegrationTest;
import denshchikov.dmitry.taskcreator.model.Task;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.dao.EmptyResultDataAccessException;

import java.util.UUID;

import static org.assertj.core.api.BDDAssertions.catchException;
import static org.assertj.core.api.BDDAssertions.then;

@SpringBootTest
public class TaskServiceTest implements IntegrationTest {

  @Autowired TaskService taskService;

  @Test
  void should_MarkTaskAsCompleted_When_TaskCreated() {
    // Given
    var taskId = UUID.randomUUID();
    var taskForCreate = new Task(taskId, "Test title", "Test description", false);
    var taskForUpdate = new Task(taskId, "Test title", "Test description", true);

    // When
    taskService.save(taskForCreate);
    taskService.update(taskForUpdate);
    var result = taskService.get(taskId);

    // Then
    then(result.isCompleted()).isTrue();
  }

  @Test
  void should_ThrowException_When_RetrievingTaskAfterDeletion() {
    // Given
    var taskId = UUID.randomUUID();
    var taskForCreate = new Task(taskId, "Test title", "Test description", false);

    // When
    taskService.save(taskForCreate);
    taskService.delete(taskId);
    var exception = catchException(() -> taskService.get(taskId));

    // Then
    then(exception).isInstanceOf(EmptyResultDataAccessException.class).hasNoCause();
  }
}

File: src/test/kotlin/denshchikov/dmitry/taskcreator/service/TaskServiceTest.kt
package denshchikov.dmitry.taskcreator.service

import denshchikov.dmitry.taskcreator.config.IntegrationTest
import denshchikov.dmitry.taskcreator.model.Task
import org.assertj.core.api.BDDAssertions.catchException
import org.assertj.core.api.BDDAssertions.then
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.dao.EmptyResultDataAccessException
import java.util.*

@SpringBootTest
internal class TaskServiceTest : IntegrationTest {

    @Autowired
    lateinit var taskService: TaskService


    @Test
    fun should_MarkTaskAsCompleted_When_TaskCreated() {
        // Given
        val taskId = UUID.randomUUID()
        val taskForCreate = Task(taskId, "Test title", "Test description")
        val taskForUpdate = Task(taskId, "Test title", "Test description", true)

        // When
        taskService.save(taskForCreate)
        taskService.update(taskForUpdate)
        val result = taskService.get(taskId)

        // Then
        then(result.isCompleted)
            .isTrue
    }

    @Test
    fun should_ThrowException_When_RetrievingTaskAfterDeletion() {
        // Given
        val taskId = UUID.randomUUID()
        val taskForCreate = Task(taskId, "Test title", "Test description")

        // When
        taskService.save(taskForCreate)
        taskService.delete(taskId)
        val exception = catchException { taskService.get(taskId) }

        // Then
        then(exception)
            .isInstanceOf(EmptyResultDataAccessException::class.java)
            .hasNoCause()
    }

}